plugins/modules/win_domain_ou.ps1

```diff
@@ -0,0 +1,257 @@
+    }
+    if ($Object.properties) {
+        if ($Object.properties.description) { $params.Description = $Object.properties.description }
+        if ($Object.properties.city) { $params.City = $Object.properties.city }
+        if ($Object.properties.state) { $params.State = $Object.properties.state }
+        if ($Object.properties.street_address) { $params.StreetAddress = $Object.properties.street_address }
+        if ($Object.properties.postal_code) { $params.PostalCode = $Object.properties.postal_code }
+        if ($Object.properties.country) { $params.Country = $Object.properties.country }
+        if ($Object.properties.managed_by) { $params.ManagedBy = $Object.properties.managed_by }
+    }
+    # convert to psobject & return
+    [PSCustomObject]$params
+}
+
+Function Get-OuObject {
+    Param([PSObject]$Object)
+    $obj = $Object | Select-Object -Property * -ExcludeProperty nTSecurityDescriptor | ConvertTo-Json -Depth 1 | ConvertFrom-Json
+    return $obj
+}
+
+# attempt import of module
+Try { Import-Module ActiveDirectory }
+Catch { $module.FailJson("The ActiveDirectory module failed to load properly: $($_.Exception.Message)", $_) }
+
+Try{
+    $all_ous = Get-ADOrganizationalUnit @extra_args
+}Catch{$module.FailJson("Get-ADOrganizationalUnit failed: $($_.Exception.Message)", $_) }
+
+# set path if not defined to base domain
+if ($null -eq $path){
+    if ($($all_ous | Measure-Object | Select-Object -ExpandProperty Count) -eq 1){
+        $matched = $all_ous.DistinguishedName -match "DC=.+"
+    }elseif ($($all_ous | Measure-Object | Select-Object -ExpandProperty Count) -gt 1) {
+        $matched = $all_ous[0].DistinguishedName -match "DC=.+"
+    }else{
+        $module.FailJson("Path was null and unable to determine default domain $($_.Exception.Message)", $_)
+    }
+    if ($matched){
+        $path = $matches.Values[0]
+    }else{
+        $module.FailJson("Unable to find default domain $($_.Exception.Message)", $_)
+    }
+}
+$module.Result.path = $path
+
+# determine if requested OU exist
+Try {
+    $current_ou = $all_ous | Where-Object {
+            $_.DistinguishedName -eq "OU=$name,$path"}
+    $module.Diff.before = $current_ou
+    $module.Result.ou = Get-OuObject $module.Diff.before
+} Catch {
+    $module.Diff.before = $null
+    $current_ou = $null
+}
+if ($state -eq "present") {
+    # ou does not exist, create object
+    if(-not $current_ou) {
+        $params.Name = $name
+        $params.Path = $path
+        $module.Result.params = $params
+        Try {
+            New-ADOrganizationalUnit @params @onboard_extra_args -ProtectedFromAccidentalDeletion $protected -WhatIf:$check_mode
+        }Catch {
+            $module.FailJson("Failed to create organizational unit: $($_.Exception.Message)", $_)
+        }
+    }
+    # ou exists, update object
+    if ($current_ou) {
+        Try {
+            Set-ADOrganizationalUnit -Identity "OU=$name,$path" @params @onboard_extra_args -WhatIf:$check_mode
```

Comment:
There are no checks here to determine whether this call is actually required or not.

Reply:
added checks